opensrp-reveal/src/main/java/org/smartregister/reveal/view/ListTasksActivity.java

```diff
@@ -213,10 +217,17 @@ public void onStyleLoaded(@NonNull Style style) {
                         geoJsonSource = style.getSourceAs(getString(R.string.reveal_datasource_name));
 
                         selectedGeoJsonSource = style.getSourceAs(getString(R.string.selected_datasource_name));
-                        RevealMapHelper.addSymbolLayers(style, ListTasksActivity.this);
+                        RevealMapHelper.addCustomLayers(style, ListTasksActivity.this);
                     }
                 });
+
                 mMapboxMap = mapboxMap;
+                mMapboxMap.addOnCameraMoveListener(new MapboxMap.OnCameraMoveListener() {
+                    @Override
+                    public void onCameraMove() {
+                        addCaseIndexBoundary(mMapboxMap, getContext());
```

Comment:
Why is this, here why not add index case when the GeoJSON Sources changes i.e one changes the Plan or OA, since there will be only one index case per Plan and OA

Reply:
This redraws with zoom.

Reply:
This is rather ineffecient calling this every time one moves/tilts/zoom the map

Reply:
That's the only way I know it resizes on zoom. Additionally, this is not an expensive operation. Do you have an alternative for redrawing on zoom?